### PR TITLE
add type definitions

### PR DESCRIPTION
**Pull request by justincorrigible**
includes type definitions with the transpiled files.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
+**/.DS_STORE
```
Comment (justincorrigible) [inline .gitignore:1]: 🤷

File: .npmignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+src/
```
Comment (justincorrigible) [inline .npmignore:1]: the source files needn't be included in the package.

File: package.json [hunk 3]
```diff
@@ -1,8 +1,9 @@
 {
-  "name": "sqon",
+  "name": "sqon-builder",
```
Comment (justincorrigible) [inline package.json:2]: for ease of publication, and to enable dev-linking locally.

File: package.json [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
   "version": "2.0.0",
   "description": "SQON creation and manipulation library.",
   "main": "dist/index.js",
+  "types": "dist/index.d.ts",
```
Comment (justincorrigible) [inline package.json:6]: this may be superfluous, but guides devs who want to manually explore the types.

File: tsconfig.json [hunk 5]
```diff
@@ -1,5 +1,6 @@
 {
   "compilerOptions": {
+    "declaration": true,
```
Comment (justincorrigible) [inline tsconfig.json:3]: this tells TSC to produce the declaration files when building.

**Review by justincorrigible — COMMENTED** [review]
blah blah blah

**Review by joneubank — APPROVED** [review]

**Pull request merged**